### PR TITLE
Test, Voice, Audio processors added for MUStARD

### PR DESCRIPTION
**Pull request by applexi**
Closes # 

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [X] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [X] All the tests have passed
- [ ] Branch name follows `type/descript` (e.g. `feature/add-llm-agents`)
- [x] Ready for code review

### CODE REVIEW

**Comment by lwaekfjlk** [comment]
3 processors: vision processor / audio processor / text processor

many many processor task: sarcasm detection task

Key component:

- [x] Processors: collect three processors for three modalities

-  [ ] Up-tree: each processor can output a prob based on softmax results (sample one hidden representation based on probability of the sooftmax results) [prob is an indicator of whether it is sarcasm or not]

   Like text 0.7, audio 0.2, vision 0.1 --> weighted output of all three processors  --> to push to the upper level

- [ ] Feedback: build a classification loss based on this representation for sarcasm detection, each processor can gain feedback based on the loss

- [ ] Link construction: Built based on similarity and cooccurrence, if their output representation is similar with each other, the link weight is the cosine simialrity between two outputs. Link = cos(x_a, c_v).

- [ ] Modality Fusion Training: After confirming that two processors are linked, we just train another 2-layer neural network for the two output of these processors to get a brand-new representation. f(x_a, x_v) = x_fusion.

- [ ] Updated inference: with link and fusion NN, we do inference on the new architecture again.
